dali/python/nvidia/dali/plugin/jax.py

```diff
@@ -50,3 +55,200 @@ def _to_jax_array(dali_tensor: TensorGPU) -> jax.Array:
+                  is called internally automatically.
+    last_batch_policy: optional, default = LastBatchPolicy.FILL
+                What to do with the last batch when there are not enough samples in the epoch
+                to fully fill it. See :meth:`nvidia.dali.plugin.base_iterator.LastBatchPolicy`
+    last_batch_padded : bool, optional, default = False
+                Whether the last batch provided by DALI is padded with the last sample
+                or it just wraps up. In the conjunction with ``last_batch_policy`` it tells
+                if the iterator returning last batch with data only partially filled with
+                data from the current epoch is dropping padding samples or samples from
+                the next epoch. If set to ``False`` next
+                epoch will end sooner as data from it was consumed but dropped. If set to
+                True next epoch would be the same length as the first one. For this to happen,
+                the option `pad_last_batch` in the reader needs to be set to True as well.
+                It is overwritten when `reader_name` argument is provided
+    prepare_first_batch : bool, optional, default = True
+                Whether DALI should buffer the first batch right after the creation of the iterator,
+                so one batch is already prepared when the iterator is prompted for the data
+
+    Example
+    -------
+    With the data set ``[1,2,3,4,5,6,7]`` and the batch size 2:
+
+    last_batch_policy = LastBatchPolicy.PARTIAL, last_batch_padded = True  -> last batch = ``[7]``,
+    next iteration will return ``[1, 2]``
+
+    last_batch_policy = LastBatchPolicy.PARTIAL, last_batch_padded = False -> last batch = ``[7]``,
+    next iteration will return ``[2, 3]``
+
+    last_batch_policy = LastBatchPolicy.FILL, last_batch_padded = True   -> last batch = ``[7, 7]``,
+    next iteration will return ``[1, 2]``
+
+    last_batch_policy = LastBatchPolicy.FILL, last_batch_padded = False  -> last batch = ``[7, 1]``,
+    next iteration will return ``[2, 3]``
+
+    last_batch_policy = LastBatchPolicy.DROP, last_batch_padded = True   -> last batch = ``[5, 6]``,
+    next iteration will return ``[1, 2]``
+
+    last_batch_policy = LastBatchPolicy.DROP, last_batch_padded = False  -> last batch = ``[5, 6]``,
+    next iteration will return ``[2, 3]``
+    """
+
+    def __init__(
+            self,
+            pipelines,
+            output_map,
+            size=-1,
+            reader_name=None,
+            auto_reset=False,
+            last_batch_padded=False,
+            last_batch_policy=LastBatchPolicy.FILL,
+            prepare_first_batch=True):
+
+        # For now this iterator supports only one pipeline. This is due to how multiple
+        # pipelines need to be handled in JAX -> return one output that is backed by
+        # multiple devices.
+        # TODO(awolant): Implement this.
+        # assert len(pipelines) == 1, \
```

Comment:
Maybe you can add assert this to enforce what is supported now?

Reply:
This is implemented now and this comment removed